includes/class-smaily-for-wp.php

```diff
@@ -138,6 +140,15 @@ private function define_public_hooks() {
 	 * @since 3.0.0
 	 */
 	public static function activate() {
+		// Run upgrade scripts before dbDelta, as the latter may delete information.
+		$plugin_version = SMLY4WP_PLUGIN_VERSION;
+		if ( $plugin_version !== get_option( 'smailyforwp_db_version' ) ) {
+			if ( version_compare( $plugin_version, '3.0.0', '=' ) ) {
+				require_once SMLY4WP_PLUGIN_PATH . '/migrations/upgrade-3-0-0.php';
```

Comment:
`SMLY4WP_PLUGIN_PATH` usage should be normalized - it has a trailing slash (this `require_once` path is `/path/to/wp-content/plugins/smaily-for-wp//migrations/upgrade-3-0-0.php`).

Reply:
e6b3b2c